go/BUILD.bazel

```diff
@@ -0,0 +1,119 @@
+load("@bazel_skylib//rules:native_binary.bzl", "native_binary")
+load("@gazelle//:def.bzl", "gazelle")
+load("@rules_pkg//pkg:install.bzl", "pkg_install")
+load("@rules_pkg//pkg:mappings.bzl", "pkg_attributes", "pkg_filegroup", "pkg_files")
+load("@rules_pkg//pkg:zip.bzl", "pkg_zip")
+load("//:defs.bzl", "codeql_platform")
+
+gazelle(
+    name = "gazelle",
+    extra_args = ["go/extractor"],
+)
+
+_gen_binaries = [
+    "@rules_go//go",
+    ":gazelle",
+    "//go/extractor/cli/go-gen-dbscheme",
+]
+
+py_binary(
+    name = "gen",
+    srcs = ["gen.py"],
+    args = ["$(rlocationpath %s)" % bin for bin in _gen_binaries],
+    data = _gen_binaries,
+    deps = ["@rules_python//python/runfiles"],
+)
+
+# this is an internal copy of the dbscheme to be used by extractor-pack
+# this allows the extractor-pack target to be independent and up-to-date with respect to
+# having run //go:gen to update the checked in files
+genrule(
+    name = "dbscheme",
+    outs = ["go.dbscheme"],
+    cmd = "$(execpath //go/extractor/cli/go-gen-dbscheme) $@",
+    tools = ["//go/extractor/cli/go-gen-dbscheme"],
+)
+
+pkg_files(
+    name = "resources",
+    srcs = [
+        "LICENSE",
+        "codeql-extractor.yml",
+        "ql/lib/go.dbscheme.stats",
+        ":dbscheme",
+    ],
+)
+
+pkg_filegroup(
+    name = "extractor-pack-generic",
+    srcs = [
+        ":resources",
+        "//go/codeql-tools",
+        "//go/downgrades",
+        "//go/extractor:tokenizer",
+    ],
+    visibility = ["//visibility:public"],
+)
+
+pkg_files(
+    name = "extractor-pack-arch",
+    srcs = [
+        "//go/extractor/cli/go-autobuilder",
+        "//go/extractor/cli/go-bootstrap",
+        "//go/extractor/cli/go-build-runner",
+        "//go/extractor/cli/go-extractor",
+        "//go/extractor/cli/go-gen-dbscheme",
+        "//go/extractor/cli/go-tokenizer",
+    ],
+    attributes = pkg_attributes(mode = "0755"),
```

Comment:
This is (partially) what we have the `dist` wrapper rules in the internal repository for. Should we move those into the `codeql` repo so that we can use them here? I think it'd make for a cleaner and more consistent experience across language packs.

I think that's the only internal dependency for one or two other language packs, too, thus allowing us to build those from the external repo, too (something I've not prioritized tbh).

Reply:
yep, I would definitely like to tackle that, but rather in a separate PR. I would also like to rethink that API:
* having a way to not give executable permissions to every file
* rename the rules to be more adherent to `rules_pkg` (`pack_zip -> codeql_pkg_files`, `dist -> codeql_pkg_filegroup`)
* give a final `codeql_pack` rule/macro encapsulating all the `generic`/`arch`/`codeql_platform` shenanigans, and with the installation target built-in
* maybe other goodies

Reply:
I put this onto the agenda for our next 1:1, so we don't forget about it entirely.

Reply:
this morning [I've jotted down what I had in mind](https://github.com/github/codeql/pull/16432) we can discuss over that :slightly_smiling_face: